src/Mailer/TransportRegistry.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         3.7.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Mailer;
+
+use BadMethodCallException;
+use Cake\Core\App;
+use Cake\Core\ObjectRegistry;
+use RuntimeException;
+
+/**
+ * An object registry for maieler transports.
+ */
+class TransportRegistry extends ObjectRegistry
+{
+    /**
+     * Resolve a mailer tranport classname.
+     *
+     * Part of the template method for Cake\Core\ObjectRegistry::load()
+     *
+     * @param string|\Cake\Mailer\AbstractTransport $class Partial classname to resolve or transport instance.
+     * @return string|false Either the correct classname or false.
+     */
+    protected function _resolveClassName($class)
+    {
+        if (is_object($class)) {
+            return $class;
+        }
+
+        $className = App::className($class, 'Mailer/Transport', 'Transport');
+
+        if (!$className) {
+            $className = App::className($class, 'Network/Email', 'Transport');
+            if ($className) {
+                trigger_error(
```

Comment:
Should this use `deprecationWarning()` now? I think this codes previous home predates that function's existence.

Reply:
Yes it should, I'll update.